### PR TITLE
Refactor/pro 1377/package id pk

### PR DESCRIPTION
**Pull request by simao**
Remove (namespace, name, version) as pk, do not use namespace where possible.

### CODE REVIEW

File: common/src/main/scala/org/genivi/sota/db/Operators.scala [hunk 1]
```diff
@@ -53,6 +53,16 @@ object Operators {
 
       DBIOOps(dbio).failIfNone(t)
     }
+
+    def failIfEmpty(t: Throwable)
```
Comment (rdanitz) [inline common/src/main/scala/org/genivi/sota/db/Operators.scala]: unused?

File: common/src/main/scala/org/genivi/sota/rest/ResponseConversions.scala [hunk 2]
```diff
@@ -1,12 +1,11 @@
...
+ * Copyright: Copyright (C) 2015, Jaguar Land Rover
  * License: MPL-2.0
  */
-package org.genivi.sota.core.data.client
+package org.genivi.sota.rest
 
-import shapeless.HList
 import shapeless.ops.function.FnToProduct
-import shapeless._
+import shapeless.{HList, _}
```
Comment (rdanitz) [inline common/src/main/scala/org/genivi/sota/rest/ResponseConversions.scala]: ;)

File: external-resolver/src/main/scala/org/genivi/sota/resolver/db/DeviceRepository.scala [hunk 3]
```diff
@@ -90,82 +92,75 @@ object DeviceRepository {
...
 
   // scalastyle:off
-  class InstalledPackageTable(tag: Tag) extends Table[(Namespace, Uuid, PackageId)](tag, "InstalledPackage") {
-    def device         = column[Uuid]("device_uuid")
-    def namespace      = column[Namespace]("namespace")
-    def packageName    = column[PackageId.Name]("packageName")
-    def packageVersion = column[PackageId.Version]("packageVersion")
+  class InstalledPackageTable(tag: Tag) extends Table[(Uuid, UUID)](tag, "InstalledPackage") {
+    def device = column[Uuid]("device_uuid")
+    def packageUuid = column[UUID]("package_uuid")
```
Comment (rdanitz) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/db/DeviceRepository.scala]: Now we're using java.util.UUID and an AnyVal wrapping a Refined[String,Uuid].
what do you think, should we unify this in the future? I had a discussion with @stevana  about datatype specific UUID's in order to not being able to mix e.g. device UUIDs with package UUIDs.
What would you say, overkill?

**Comment by rdanitz** [comment]
otherwise, lgtm

**Comment by rdanitz** [comment]
review score: 11/11

File: external-resolver/src/main/scala/org/genivi/sota/resolver/db/DeviceRepository.scala (hunk 3 above)
Comment (simao) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/db/DeviceRepository.scala]: Having all those different alias and types feels like overkill sometimes so I am not sure about that. Anyway we should definitely unify all these whatever we choose.

**Pull request merged**